migrations/versions/0d4e564b1876_backfill_distribution.py

```diff
@@ -0,0 +1,186 @@
+                "-",
+                machine_table.c.cpu_core_count,
+                "-",
+                machine_table.c.cpu_thread_count,
+                "-",
+                machine_table.c.memory_bytes,
+            ).label("hash"),
+            func.max(summary_table.c.unit).label("unit"),
+            func.avg(summary_table.c.mean).label("mean_mean"),
+            func.stddev(summary_table.c.mean).label("mean_sd"),
+            func.avg(summary_table.c.min).label("min_mean"),
+            func.stddev(summary_table.c.min).label("min_sd"),
+            func.avg(summary_table.c.max).label("max_mean"),
+            func.stddev(summary_table.c.max).label("max_sd"),
+            func.avg(summary_table.c.median).label("median_mean"),
+            func.stddev(summary_table.c.median).label("median_sd"),
+            func.min(commits_up.c.timestamp).label("first_timestamp"),
+            func.max(commits_up.c.timestamp).label("last_timestamp"),
+            func.count(summary_table.c.mean).label("observations"),
+        )
+        .group_by(
+            summary_table.c.case_id,
+            summary_table.c.context_id,
+            summary_table.c.machine_id,
+            machine_table.c.name,
+            machine_table.c.cpu_core_count,
+            machine_table.c.cpu_thread_count,
+            machine_table.c.memory_bytes,
+        )
+        .join(run_table, run_table.c.id == summary_table.c.run_id)
+        .join(machine_table, machine_table.c.id == summary_table.c.machine_id)
+        .join(commits_up, commits_up.c.id == run_table.c.commit_id)
+        .filter(
+            run_table.c.name.like("commit: %"),
+            summary_table.c.case_id == case_id,
+            summary_table.c.context_id == context_id,
+            func.concat(
+                machine_table.c.name,
+                "-",
+                machine_table.c.cpu_core_count,
+                "-",
+                machine_table.c.cpu_thread_count,
+                "-",
+                machine_table.c.memory_bytes,
+            )
+            == machine_hash,
+        )
+    )
+
+
+def upgrade():
+    connection = op.get_bind()
+    meta = MetaData()
+    meta.reflect(bind=connection)
+
+    commit_table = meta.tables["commit"]
+    distribution_table = meta.tables["distribution"]
+    machine_table = meta.tables["machine"]
+    run_table = meta.tables["run"]
+    summary_table = meta.tables["summary"]
+
+    runs = connection.execute(run_table.select())
+    commits = connection.execute(commit_table.select())
+    distributions = connection.execute(distribution_table.select())
+    machines = connection.execute(machine_table.select())
+    runs_by_id = {r["id"]: r for r in runs}
+    commits_by_id = {c["id"]: c for c in commits}
+    machines_by_id = {m["id"]: m for m in machines}
+
+    seen = {
+        f'{d["sha"]}{d["case_id"]}{d["context_id"]}{d["machine_id"]}'
+        for d in distributions
+    }
+
+    summaries = connection.execute(summary_table.select())
```

Comment:
@dianaclarke If we only select summary rows with run_ids for runs with name like `commit: %`, we can cut down loop iteration (`for summary in summaries`) by more than 50%:


```
select count(1) 
from summary

2288167

select count(1) 
from summary, machine
where summary.machine_id = machine.id and
machine.name in ('ursa-i9-9960x', 'ec2-t3-large-us-east-2', 'ec2-t3-xlarge-us-east-2', 'ursa-thinkcentre-m75q') 

1525400


select count(1) 
from summary, machine, run
where summary.machine_id = machine.id and
machine.name in ('ursa-i9-9960x', 'ec2-t3-large-us-east-2', 'ec2-t3-xlarge-us-east-2', 'ursa-thinkcentre-m75q') and
summary.run_id = run.id and
run.name like 'commit: %'

993577


select count(1) 
from summary, run
where summary.run_id = run.id and
run.name like 'commit: %'

1036257
```